### PR TITLE
[SPARK-50612][SQL] Normalize ordering in the project list of an inner Project 

### PR DESCRIPTION
**Pull request by mihailotim-db**
### What changes were proposed in this pull request?
Normalize ordering in the project list of an inner Project. For example, following two plans should be the same after normalization:
```
Project[col1]                                  Project[col1]
:- Project[col1, col2]                         :- Project[col2, col1]
```
This is because only the top most ordering matters for comparing plans and inner project list can be computed in a non-deterministic way (i.e. via `HashSet`/`HashMap`)

### Why are the changes needed?

For plan comparison in the single-pass resolver.

### How was this patch tested?
Added a new test suite.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/NormalizePlanSuite.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  test("Normalize ordering in a project list of an inner Project") {
+    val baselinePlan =
+      LocalRelation($"col1".int, $"col2".string).select($"col1", $"col2").select($"col1")
+    val testPlan =
+      LocalRelation($"col1".int, $"col2".string).select($"col2", $"col1").select($"col1")
+
+    assert(baselinePlan != testPlan)
+    assert(NormalizePlan(baselinePlan) == (NormalizePlan(testPlan)))
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/NormalizePlanSuite.scala:44]: nit:
```suggestion
    assert(NormalizePlan(baselinePlan) == NormalizePlan(testPlan))
```

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/NormalizePlanSuite.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/NormalizePlanSuite.scala:46]:
```suggestion
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/NormalizePlan.scala [hunk 3]
```diff
@@ -95,19 +95,27 @@ object NormalizePlan extends PredicateHelper {
...
-            }
-          }
-          .asInstanceOf[Seq[NamedExpression]]
-        Project(projList, child)
+        Project(normalizeProjectList(projectList), child)
       case c: KeepAnalyzedQuery => c.storeAnalyzedQuery()
     }
   }
 
+  private def normalizeProjectList(projectList: Seq[NamedExpression]): Seq[NamedExpression] =
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/NormalizePlan.scala:110]: Please, add curly braces around the method body, see https://github.com/databricks/scala-style-guide?tab=readme-ov-file#curly-braces

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/NormalizePlanSuite.scala (hunk 1 above)
Comment (mihailotim-db) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/NormalizePlanSuite.scala:44]: Nice spot

**Comment by mihailotim-db** [comment]
@MaxGekk I addressed all comments. Failures seem unrelated, can you please take a look?

**Review by MaxGekk — APPROVED** [review]

**Comment by mihailotim-db** [comment]
@MaxGekk CI has passed, we can merge it now

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @mihailotim-db.

**Pull request closed without merging**